design-toolkit/components/src/components/notice/index.tsx

```diff
@@ -0,0 +1,318 @@
+                {...primary}
+                size={size}
+                onPress={() => {
+                  onPrimaryAction?.();
+
+                  if (shouldCloseOnAction) {
+                    onClose?.();
+                  }
+                }}
+              />
+            )}
+            {secondary && (
+              <Button
+                variant='outline'
+                color={ButtonColorMap[color]}
+                {...secondary}
+                size={size}
+                onPress={() => {
+                  onSecondaryAction?.();
+
+                  if (shouldCloseOnAction) {
+                    onClose?.();
+                  }
+                }}
+              />
+            )}
+          </div>
+        )}
+        {showClose && (
+          <div className={closeContainer()}>
+            <Button
+              color={ButtonColorMap[color]}
+              variant='icon'
+              onPress={onClose}
+            >
+              <Icon>
+                <Cancel />
+              </Icon>
+            </Button>
+          </div>
+        )}
+      </ToastContent>
+    </Toast>
+  );
+}
+
+function matchesMetadata(
+  payload: Record<string, unknown>,
+  metadata?: Record<string, unknown>,
+) {
+  if (!metadata) {
+    return false;
+  }
+
+  return Object.entries(payload).every(
+    ([key, value]) => key in metadata && isEqual(metadata[key], value),
+  );
+}
+
+function NoticeList({
+  id,
+  classNames,
+  defaultColor,
+  defaultTimeout,
+  hideClearAll,
+  limit = 3,
+  global,
+  placement,
+  size = 'medium',
+  ...rest
+}: NoticeListProps) {
+  const queue = useMemo(
+    () => new ToastQueue<NoticeContent>({ maxVisibleToasts: limit }),
+    [limit],
+  );
+  const state = useToastQueue(queue);
+
+  const [hasNotices, setHasNotices] = useState(false);
+  const ref = createRef<HTMLDivElement>();
+  const { regionProps } = useToastRegion(rest, queue, ref);
+
+  const { useEmit, useOn } = useBus<
+    NoticeQueueEvent | NoticeDequeueEvent | NoticeActionEvent
+  >();
+  const emitActionPrimary = useEmit(NoticeEventTypes.actionPrimary);
+  const emitActionSecondary = useEmit(NoticeEventTypes.actionSecondary);
+  const emitClose = useEmit(NoticeEventTypes.close);
+
+  useOn(NoticeEventTypes.queue, (data) => {
+    if ((id && data.payload.target === id) || !id) {
+      queue.add(
+        {
+          ...data.payload,
+          id: data.payload.id || uuid(),
+          color: defaultColor || data.payload.color,
+        },
+        {
+          timeout: defaultTimeout || data.payload.timeout,
+        },
+      );
+    }
+  });
+
+  useOn(NoticeEventTypes.dequeue, (data) => {
+    // @ts-expect-error : queue.queue exists, but is not currently documented
+    const dequeue = queue.queue.filter((toast: QueuedToast<NoticeContent>) => {
+      return matchesMetadata(data.payload, toast.content);
+    });
+
+    // @ts-expect-error : queue.queue exists, but is not currently documented
```

Comment:
Can you consolidate the accessor to a single instance to minimize the number of `@ts-expect-error`s please?